### PR TITLE
Extract a Role class from Plans

### PR DESCRIPTION
**Pull request by Jasopaum**

### CODE REVIEW

File: syft/execution/role.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+            return None
+
+    def add_tensor_to_state(self, tensor):
+        placeholder = sy.PlaceHolder(id=tensor.id, owner=self.owner)
+        placeholder.instantiate(tensor)
+        self.state.state_placeholders.append(placeholder)
+        # TODO isn't it weird that state placeholders are both in state and plan?
+        self.placeholders[tensor.id] = placeholder
+
+    def fecth_placeholders_from_ids(self, obj):
```
Comment (iamtrask) [inline syft/execution/role.py]: typo fetch_placeholders_from_ids

Comment (Jasopaum) [reply syft/execution/role.py]: Thanks! Just fixed that

File: syft/generic/object.py [hunk 2]
```diff
@@ -17,7 +17,7 @@ def __init__(
         self,
         id: int = None,
         owner: "sy.workers.AbstractWorker" = None,
-        tags: List[str] = None,
+        tags: Set[str] = None,
```
Comment (karlhigley) [inline syft/generic/object.py]: 💯

File: syft/execution/plan.py [hunk 3]
```diff
@@ -582,42 +368,39 @@ def __str__(self):
...
+        #         f"_{extract_tag(arg)}" if isinstance(arg, PlaceHolder) else str(arg)
+        #         for arg in action.args
+        #     )
+        #     if action.kwargs:
+        #         line += ", " + ", ".join(f"{k}={w}" for k, w in action.kwargs.items())
+        #     line += ")\n"
+        #     out += line
+
+        # out += "    return "
+        # out += ", ".join(f"_{extract_tag(p)}" for p in self.find_placeholders("output"))
```
Comment (karlhigley) [inline syft/execution/plan.py]: Can we remove this now? 👍

Comment (Jasopaum) [reply syft/execution/plan.py]: I want to because I don't find it very useful to print a list of actions like that... What do you think?

**Pull request merged**